packages/alfa-aria/src/node.ts

```diff
@@ -401,59 +406,51 @@ export namespace Node {
 
         const style = Style.from(node, device);
 
-        // Elements that are not rendered at all by means of `display: none`
-        // are never exposed in the accessibility tree, nor are their
-        // descendants.
+        // Elements that are not rendered at all by means are never exposed in
+        // the accessibility tree, nor are their descendants.
         //
-        // As we're building the accessibility tree top-down, we only need to
-        // check the element itself for `display: none` and can safely
-        // disregard its ancestors as they will already have been checked.
-        if (
-          style
-            .computed("display")
-            .some(({ values: [outside] }) => outside.value === "none")
-        ) {
+        // Since `aria-owns` can create an accessibility tree that is fairly
+        // different from the DOM tree, but being rendered is a property of the
+        // DOM, we may "jump" onto a node which is not rendered due to some DOM
+        // ancestor (so, unknowingly of the current accessibility tree traversal).
+        // Therefore, we cannot just look at some property of the current node.
+        //
+        // Since `isRendered` is cached, and evaluating it is needed for almost
+        // all nodes in the DOM, this is inexpensive.
+        if (test(not(isRendered(device)), node)) {
           return Inert.of(node);
         }
 
         let children: (state: State) => Iterable<Node>;
 
-        // Children of <iframe> elements act as fallback content in legacy user
-        // agents and should therefore never be included in the accessibility
-        // tree.
-        if (node.name === "iframe") {
-          children = () => [];
-        }
```

Comment:
Since `isRendered` already includes `isFallback`, this test is not needed anymore.
This does change result a bit by returning a list of `Inert` children (rather than an empty one), which is likely OK.